File: components_page/components/button_group/radios.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

button_group <- htmlDiv(
  list(
    dbcRadioItems(
      id = "radios",
      className = "btn-group",
      inputClassName = "btn-check",
      labelClassName = "btn btn-outline-primary",
      labelCheckedClassName = "active",
      options = list(
        list(label = "Option 1", value = 1),
        list(label = "Option 2", value = 2),
        list(label = "Option 3", value = 3)
      ),
      value = 1,
    ),
    htmlDiv(id = "output")
  ),
  className = "radio-group",
)


app$callback(
  output("output", "children"),
  list(input("radios", "value")),
  function(value) {
    return(sprintf("Selected value: %d", value))
  }
)

File: components_page/components/button_group/radios.py
import dash_bootstrap_components as dbc
from dash import Input, Output, html

button_group = html.Div(
    [
        dbc.RadioItems(
            id="radios",
            className="btn-group",
            inputClassName="btn-check",
            labelClassName="btn btn-outline-primary",
            labelCheckedClassName="active",
            options=[
                {"label": "Option 1", "value": 1},
                {"label": "Option 2", "value": 2},
                {"label": "Option 3", "value": 3},
            ],
            value=1,
        ),
        html.Div(id="output"),
    ],
    className="radio-group",
)


@app.callback(Output("output", "children"), [Input("radios", "value")])
def display_value(value):
    return f"Selected value: {value}"
